homeassistant/components/drop_connect/sensor.py

```diff
@@ -0,0 +1,432 @@
+
+    def __init__(self, device) -> None:
+        """Initialize the high system pressure today sensor."""
+        super().__init__(self._attr_translation_key, device)
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the high system pressure today."""
+        if self._device.high_system_pressure is None:
+            return None
+        return round(self._device.high_system_pressure, 1)
+
+
+class DROP_LowSystemPressureSensor(DROP_Entity, SensorEntity):
+    """Monitors the low system pressure today."""
+
+    _attr_device_class = SensorDeviceClass.PRESSURE
+    _attr_native_unit_of_measurement = "psi"
+    _attr_translation_key = "low_system_pressure"
+
+    def __init__(self, device) -> None:
+        """Initialize the low system pressure today sensor."""
+        super().__init__(self._attr_translation_key, device)
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the low system pressure today."""
+        if self._device.low_system_pressure is None:
+            return None
+        return round(self._device.low_system_pressure, 1)
+
+
+class DROP_BatterySensor(DROP_Entity, SensorEntity):
+    """Monitors the battery level."""
+
+    _attr_device_class = SensorDeviceClass.BATTERY
+    _attr_native_unit_of_measurement = "%"
+    _attr_translation_key = "battery"
+
+    def __init__(self, device) -> None:
+        """Initialize the battery sensor."""
+        super().__init__(self._attr_translation_key, device)
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the battery level."""
+        if self._device.battery is None:
+            return None
+        return round(self._device.battery, 1)
+
+
+class DROP_TemperatureSensorF(DROP_Entity, SensorEntity):
+    """Monitors the temperature."""
+
+    _attr_device_class = SensorDeviceClass.TEMPERATURE
+    _attr_native_unit_of_measurement = "°F"
+    _attr_translation_key = "temperature_f"
+
+    def __init__(self, device) -> None:
+        """Initialize the temperature sensor."""
+        super().__init__(self._attr_translation_key, device)
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the temperature."""
+        if self._device.temperature_f is None:
+            return None
+        return round(self._device.temperature_f, 1)
+
+
+class DROP_TemperatureSensorC(DROP_Entity, SensorEntity):
+    """Monitors the temperature."""
+
+    _attr_device_class = SensorDeviceClass.TEMPERATURE
+    _attr_native_unit_of_measurement = "°C"
+    _attr_translation_key = "temperature_c"
+
+    def __init__(self, device) -> None:
+        """Initialize the temperature sensor."""
+        super().__init__(self._attr_translation_key, device)
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the temperature."""
+        if self._device.temperature_c is None:
+            return None
+        return round(self._device.temperature_c, 1)
```

Comment:
Entities should not be specific to an unit of measurement. HA has logic to convert entities to different units. Pick one of the two (the most accurate/the one you expect will be used the most/whatever) and stick with that.

Reply:
The duplication was removed in 8bef01bba0902bbc74752bbafd9eb0b9d9a1631f.